cvat/apps/redis_handler/views.py

```diff
@@ -0,0 +1,272 @@
+        "subresource": SchemaField("string"),
+        "format": SchemaField("string"),
+        "org": SchemaField("string"),
+    }
+
+    def get_queryset(self):
+        return None
+
+    @property
+    def queues(self) -> Iterable[DjangoRQ]:
+        return (django_rq.get_queue(queue_name) for queue_name in self.SUPPORTED_QUEUES)
+
+    @classmethod
+    def get_parsed_id_class(cls, queue_name: str) -> type[RequestId]:
+        return cls.PARSED_JOB_ID_CLASSES.get(queue_name, RequestId)
+
+    def _get_rq_jobs_from_queue(self, queue: DjangoRQ, user_id: int) -> list[RQJob]:
+        job_ids = set(
+            queue.get_job_ids()
+            + queue.started_job_registry.get_job_ids()
+            + queue.finished_job_registry.get_job_ids()
+            + queue.failed_job_registry.get_job_ids()
+            + queue.deferred_job_registry.get_job_ids()
+        )
+        jobs = []
+
+        ParsedIdClass = self.get_parsed_id_class(queue.name)
+
+        for job in queue.job_class.fetch_many(job_ids, queue.connection):
+            if job and is_rq_job_owner(job, user_id):
+                try:
+                    parsed_request_id = ParsedIdClass.parse(job.id)
+                except Exception:  # nosec B112
+                    continue
+
+                job.parsed_id = parsed_request_id
+                jobs.append(job)
+
+        return jobs
+
+    def _get_rq_jobs(self, user_id: int) -> list[RQJob]:
+        """
+        Get all RQ jobs for a specific user and return them as a list of RQJob objects.
+
+        Parameters:
+            user_id (int): The ID of the user for whom to retrieve jobs.
+
+        Returns:
+            List[RQJob]: A list of RQJob objects representing all jobs for the specified user.
+        """
+        all_jobs = []
+        for queue in self.queues:
+            jobs = self._get_rq_jobs_from_queue(queue, user_id)
+            all_jobs.extend(jobs)
+
+        return all_jobs
+
+    def _get_rq_job_by_id(self, rq_id: str) -> RQJob | None:
+        """
+        Get a RQJob by its ID from the queues.
+
+        Args:
+            rq_id (str): The ID of the RQJob to retrieve.
+
+        Returns:
+            Optional[RQJob]: The retrieved RQJob, or None if not found.
+        """
+        try:
+            parsed_request_id = RequestId.parse(rq_id)
+        except Exception:
+            return None
+
+        job: CustomRQJob | None = None
+
+        if parsed_request_id.queue not in self.SUPPORTED_QUEUES:
+            raise ValidationError("Unsupported queue")
+
+        queue: DjangoRQ = django_rq.get_queue(parsed_request_id.queue)
+
+        job = queue.fetch_job(rq_id)
+        if job:
+            ParsedIdClass = self.get_parsed_id_class(queue.name)
+            if (
+                type(parsed_request_id) is not ParsedIdClass # fmt: skip # pylint: disable=unidiomatic-typecheck
+            ):
+                parsed_request_id = parsed_request_id.convert_to(ParsedIdClass)
```

Comment:
@SpecLad, I remembered why we need the `convert_to` method - we must use base class  `RequestId` initially to define the queue in which an RQ job is handled. So we cannot use directly specific `RequestId` class.

Reply:
I see. But couldn't the logic for determining the appropriate subclass be moved to `RequestId.parse`? It only requires the queue name, and `RequestId.parse` knows it.